lib/permutations.rb

```diff
@@ -1,4 +1,23 @@
+# time complexity => O(n + m)
+# Space complexity => O(n) ?
 
 def permutations?(string1, string2)
-  raise NotImplementedError, "permutations? not implemented" 
+  return false if string1.length != string2.length
+
+  hash = {}
+
+  string1.each_char do |char|
+    hash[char] = true
```

Comment:
What about instead of using true and false, you count the number of each letters in string1 and then count down in the hash with string2.